types/src/blob.rs

```diff
@@ -1,60 +1,44 @@
-use celestia_proto::tendermint::types as tmtypes;
-use serde::{Deserialize, Deserializer, Serialize};
+use serde::{Deserialize, Serialize};
+use tendermint_proto::v0_34::types::Blob as RawBlob;
+use tendermint_proto::Protobuf;
 
+use crate::nmt::Namespace;
+use crate::{Bytes, Error};
+
+// NOTE: We do not the `serce(try_from)` pattern for this type
+// becase JSON representation needs to have `commitment` field but
+// Protobuf definition doesn't.
```

Comment:
I'm not sure why is that preventing the `try_from` and `into`, the Go impl seems to be creating the commitment on blob creation, so we could have that in `try_from` and be consistent with this pattern. You can leave a todo for creating a valid commitment and just put an empty vec / zeroes there

Reply:
ahh true